Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -155,14 +155,32 @@ template <class Type> class TableColumn : public API::Column {
    * is throw. In case of an overflow boost::numeric::positive_overflow or
    * boost::numeric::negative_overflow
    * is throw.
-   * @param i :: The index to an element.
+   * @param value :: The value of the element.
    */
-  double toDouble(size_t i) const override {
+  template <typename T> double convertToDouble(const T &value) const {
     typedef
         typename boost::mpl::if_c<boost::is_convertible<double, Type>::value,
                                   Type, InconvertibleToDoubleType>::type
             DoubleType;
-    return boost::numeric_cast<double, DoubleType>(m_data[i]);
+    return boost::numeric_cast<double, DoubleType>(value);
+  }
+
+  /**
+   * Cast an string to double if possible. If it's impossible
+   * boost::numeric::bad_lexical_cast
+   * is throw. In case of an overflow boost::numeric::positive_overflow or
+   * boost::numeric::negative_overflow
+   * is throw.
+   * @param value :: The value of the element.
+   */
+
+  double convertToDouble(const std::string value) const {
```

Comment:
This is better but we're still missing the reference qualifier.

Also, please remove the commented out lexical_cast.